src/util/TokenStreamBase.js

```diff
@@ -16,7 +16,7 @@ function TokenStreamBase(input, tokenData){
      * @property _reader
      * @private
      */
-    this._reader = input ? new StringReader(input.toString()) : null;
+    this._reader = input !== undefined ? new StringReader(input.toString()) : null;
```

Comment:
Or maybe it would be better to say:
```
this._reader = new StringReader(input ? input.toString() : '');
```
since I think we've established that we're just going to throw a strange exception if `this._reader === null`, and we probably don't ever want to do that.